cloud/services/subnets/subnets_test.go

```diff
@@ -199,101 +309,177 @@ func TestReconcileSubnets(t *testing.T) {
+				m.Delete(context.TODO(), "", "my-vnet", "my-subnet-1")
 			},
 		},
 		{
-			name: "skip delete if vnet is managed",
-			subnetSpec: Spec{
-				Name:                "my-subnet",
-				CIDR:                "10.0.0.0/16",
-				VnetName:            "custom-vnet",
-				RouteTableName:      "my-subnet_route_table",
-				SecurityGroupName:   "my-sg",
-				Role:                infrav1.SubnetNode,
-				InternalLBIPAddress: "10.0.0.10",
+			name:          "skip delete if vnet is managed",
+			expectedError: "",
+			expect: func(s *mock_subnets.MockSubnetScopeMockRecorder, m *mock_subnets.MockClientMockRecorder,
+				mRouteTables *mock_routetables.MockClientMockRecorder, mSecurityGroups *mock_securitygroups.MockClientMockRecorder) {
+				s.V(gomock.AssignableToTypeOf(2)).AnyTimes().Return(klogr.New())
+				s.SubnetSpecs().Return([]azure.SubnetSpec{
+					{
+						Name:                "my-subnet",
+						CIDR:                "10.0.0.0/16",
+						VNetName:            "custom-vnet",
+						RouteTableName:      "my-subnet_route_table",
+						SecurityGroupName:   "my-sg",
+						Role:                infrav1.SubnetNode,
+						InternalLBIPAddress: "10.0.0.10",
+					},
+				})
+				s.Vnet().AnyTimes().Return(&infrav1.VnetSpec{ResourceGroup: "custom-vnet-rg", Name: "custom-vnet", ID: "id1"})
+				s.ClusterName().AnyTimes().Return("fake-cluster")
+				s.ResourceGroup().AnyTimes().Return("my-rg")
+			},
+		},
+		{
+			name:          "fail delete subnet",
+			expectedError: "failed to delete subnet my-subnet in resource group my-rg: #: Internal Server Error: StatusCode=500",
+			expect: func(s *mock_subnets.MockSubnetScopeMockRecorder, m *mock_subnets.MockClientMockRecorder,
+				mRouteTables *mock_routetables.MockClientMockRecorder, mSecurityGroups *mock_securitygroups.MockClientMockRecorder) {
+				s.V(gomock.AssignableToTypeOf(2)).AnyTimes().Return(klogr.New())
+				s.SubnetSpecs().Return([]azure.SubnetSpec{
+					{
+						Name:                "my-subnet",
+						CIDR:                "10.0.0.0/16",
+						VNetName:            "my-vnet",
+						RouteTableName:      "my-subnet_route_table",
+						SecurityGroupName:   "my-sg",
+						Role:                infrav1.SubnetNode,
+						InternalLBIPAddress: "10.0.0.10",
+					},
+				})
+				s.Vnet().AnyTimes().Return(&infrav1.VnetSpec{Name: "my-vnet", ResourceGroup: "my-rg"})
+				s.ClusterName().AnyTimes().Return("fake-cluster")
+				s.ResourceGroup().AnyTimes().Return("my-rg")
+				m.Delete(context.TODO(), "my-rg", "my-vnet", "my-subnet").Return(autorest.NewErrorWithResponse("", "", &http.Response{StatusCode: 500}, "Internal Server Error"))
 			},
-			vnetSpec: &infrav1.VnetSpec{ResourceGroup: "custom-vnet-rg", Name: "custom-vnet", ID: "id1"},
-			expect:   func(m *mock_subnets.MockClientMockRecorder) {},
 		},
 	}
 	for _, tc := range testcases {
 		t.Run(tc.name, func(t *testing.T) {
 			g := NewWithT(t)
 
+			t.Parallel()
```

Comment:
this needs `tc := tc` above (see https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/806)

Reply:
@CecileRobertMichon done